In [1]:
import pandas as pd
import numpy as np
import matplotlib as plt

In [2]:
LS= pd.read_csv("C:/Users/shishf/Desktop/India LokSabha Analysis/LS.csv")

In [3]:
LS.head(3)

STATE CONSTITUENCY            NAME  WINNER PARTY SYMBOL GENDER  \
0  Telangana     ADILABAD  SOYAM BAPU RAO       1   BJP  Lotus   MALE   
1  Telangana     ADILABAD    Godam Nagesh       0   TRS    Car   MALE   
2  Telangana     ADILABAD   RATHOD RAMESH       0   INC   Hand   MALE   

   CRIMINAL_CASES   AGE CATEGORY      EDUCATION         ASSETS  LIABILITIES  \
0            52.0  52.0       ST      12th Pass    30,99,414\n  2,31,450\n    
1             0.0  54.0       ST  Post Graduate  1,84,77,888\n  8,47,000\n    
2             3.0  52.0       ST      12th Pass  3,64,91,000\n  1,53,00,000   

   GENERAL_VOTES  POSTAL_VOTES  TOTAL_VOTES  \
0         376892           482       377374   
1         318665           149       318814   
2         314057           181       314238   

   OVER_TOTAL_ELECTORS_IN_CONSTITUENCY  \
0                            25.330684   
1                            21.399929   
2                            21.092771   

   OVER_TOTAL_VOTES_POLLED _IN_CONSTITUENCY  TOTAL_ELECTORS  
0                                 35.468248         1489790  
1                                 29.964370         1489790  
2                                 29.534285         1489790

In [4]:
#No. of rows and columns in a DataFrame
LS.shape

(2263, 19)

In [5]:
LS.describe()

WINNER  CRIMINAL_CASES          AGE  GENERAL_VOTES  POSTAL_VOTES  \
count  2263.000000     1996.000000  2018.000000   2.263000e+03   2263.000000   
mean      0.238179        1.469940    52.273538   2.615991e+05    990.710561   
std       0.426064        7.677432    11.869373   2.549906e+05   1602.839174   
min       0.000000        0.000000    25.000000   1.339000e+03      0.000000   
25%       0.000000        0.000000    43.250000   2.103450e+04     57.000000   
50%       0.000000        0.000000    52.000000   1.539340e+05    316.000000   
75%       0.000000        1.000000    61.000000   4.858040e+05   1385.000000   
max       1.000000      240.000000    86.000000   1.066824e+06  19367.000000   

        TOTAL_VOTES  OVER_TOTAL_ELECTORS_IN_CONSTITUENCY  \
count  2.263000e+03                          2263.000000   
mean   2.625898e+05                            15.811412   
std    2.559822e+05                            14.962861   
min    1.342000e+03                             0.097941   
25%    2.116250e+04                             1.296518   
50%    1.544890e+05                            10.510553   
75%    4.872315e+05                            29.468185   
max    1.068569e+06                            51.951012   

       OVER_TOTAL_VOTES_POLLED _IN_CONSTITUENCY  TOTAL_ELECTORS  
count                               2263.000000    2.263000e+03  
mean                                  23.190525    1.658016e+06  
std                                   21.564758    3.145187e+05  
min                                    1.000039    5.518900e+04  
25%                                    1.899502    1.530014e+06  
50%                                   16.221721    1.679030e+06  
75%                                   42.590233    1.816857e+06  
max                                   74.411856    3.150313e+06

<h2> From the above cell, we can infer that the following points </h2>
<b><ul>
    <li> Average Age of contestant is ~52yrs, where as the youngest candidates is 25yrs old and oldest candidate is of 86 yrs </li>
    <li> out of all candidates,1996 candidates have declared their criminal records,the cases can vary from 0-240 (min-max value). We have atleast 1 person who has whooping 240 cases on his/her name</li>
    <li>Remaining 267 records have NA values in criminal column</li>
    </ul></b> 
   

In [6]:
#Names of rows and columns along with Datatypes , non-NA count
LS.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2263 entries, 0 to 2262
Data columns (total 19 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   STATE                                     2263 non-null   object 
 1   CONSTITUENCY                              2263 non-null   object 
 2   NAME                                      2263 non-null   object 
 3   WINNER                                    2263 non-null   int64  
 4   PARTY                                     2263 non-null   object 
 5   SYMBOL                                    2018 non-null   object 
 6   GENDER                                    2018 non-null   object 
 7   CRIMINAL_CASES                            1996 non-null   float64
 8   AGE                                       2018 non-null   float64
 9   CATEGORY                                  2018 non-null   object 
 10  EDUCATION                           

<h5> So we have total of 2263 records in this dataset with 19 columns .... From info() status we can see that , ~ 270 persons have not/declared their Assets, Liabilities and criminal cases </h5>

In [8]:
##### Standardizing column names to lower case
LS.columns=LS.columns.str.lower()
LS.columns

Index(['state', 'constituency', 'name', 'winner', 'party', 'symbol', 'gender',
       'criminal_cases', 'age', 'category', 'education', 'assets',
       'liabilities', 'general_votes', 'postal_votes', 'total_votes',
       'over_total_electors_in_constituency',
       'over_total_votes_polled _in_constituency', 'total_electors'],
      dtype='object')

In [9]:
#### Total Seats or Total Winners
LS["winner"].value_counts()

0    1724
1     539
Name: winner, dtype: int64

In [10]:
### Average Age of candidates Party-Wise
#SADM, JKN, VPI has the candidates with average age >=70
LS[['party', 'age']].groupby('party').mean('age').sort_values('age', ascending=0) #mean() excludes all NA values, so no need of na.rm

age
party       
SAD(M)  74.0
JKN     73.0
VPI     70.0
BOPF    69.0
ravp    68.0
...      ...
PHJSP   28.0
MOSP    27.0
CPIM    27.0
JHP     27.0
NOTA     NaN

[133 rows x 1 columns]

<h3> <i> No. of Seats won by each party</i> </h3>

In [11]:
LS[["party", "winner"]].groupby('party').agg ({'winner':[('seats_won','sum'),('seats_contested', 'count')]})

winner                
      seats_won seats_contested
party                          
AAM           0               2
AAP           1              25
ABGP          0               1
ABSKP         0               1
ADAL          2               2
...         ...             ...
WAP           0               1
WPOI          0               1
YKP           0               1
YSRCP        22              25
ravp          0               1

[133 rows x 2 columns]

<h3> <i> No. of Seats won by State </i> </h3>

In [12]:
LS[['state', 'winner']].groupby('state').agg({'winner':[('Total Seats','sum')]})

winner
                          Total Seats
state                                
Andaman & Nicobar Islands           1
Andhra Pradesh                     25
Arunachal Pradesh                   2
Assam                              14
Bihar                              41
Chandigarh                          1
Chhattisgarh                       11
Dadra & Nagar Haveli                1
Daman & Diu                         1
Goa                                 2
Gujarat                            26
Haryana                            10
Himachal Pradesh                    4
Jammu & Kashmir                     6
Jharkhand                          14
Karnataka                          28
Kerala                             20
Lakshadweep                         1
Madhya Pradesh                     29
Maharashtra                        47
Manipur                             2
Meghalaya                           2
Mizoram                             1
NCT OF Delhi                        7
Nagaland                            1
Odisha                             20
Puducherry                          1
Punjab                             13
Rajasthan                          25
Sikkim                              1
Tamil Nadu                         38
Telangana                          17
Tripura                             2
Uttar Pradesh                      78
Uttarakhand                         5
West Bengal                        42